### PR TITLE
Http disruptor

### PR DESCRIPTION
**Pull request by pablochacin**
Implement functionalities for disrupting http traffic served by a pod.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -111,13 +111,27 @@ Methods:
...
       options: controls the delay attack
         - delay: average delay in network packages (in milliseconds)
         - variation: variation in the delay (in milliseconds)
         - duration: duration of the disruption 
 
+`disruptHttp`: disrupts http requests to the pod.
+
+      Parameters:
+        options: controls the attack
+          - delay: average delay in network packages (in milliseconds)
```
Comment (dgzlopes) [inline README.md:126]:
```suggestion
          - delay: average delay in requests (in milliseconds)
```

File: README.md [hunk 2]
```diff
@@ -111,13 +111,27 @@ Methods:
...
+          - error_rate: rate of requests that will return an error
+          - error_code: error code to return
+          - duration: duration of the disruption
+          - target: port on which the requests will be intercepted (defaults to 80)
+          - port: port the transparent proxy will use to listen for requests (defaults to 8080)
+          - interface: interface on thich the traffic will be intercepted (defaults to eth0)
+
 ## Examples
 
 The [./exaples](./examples) forlder contains examples of using `k6-chaos`.
```
Comment (dgzlopes) [inline README.md:137]:
```suggestion
The [./examples](./examples) folder contains examples of using `k6-chaos`. 
```

File: README.md [hunk 3]
```diff
@@ -167,6 +181,25 @@ delay ✓ [======================================] 1 VUs        00m10.0s/10m0s
...
 
+## Introduce disruptions in http connections to pod
+
+The [disrupt-http.js example](examples/disrupt-http.js) shows how `PodDisruptor` can be used for testing the effect of disruptions in the http requests served by an instance of a deloyment. The example uses a deployment of the `httpbin`. The initial setup makes each request to take `100ms`. When executed without any disruption the statistics looks like this.
+
+```
+http_req_duration..............: avg=101.66ms min=100.59ms med=101.69ms max=107.24ms p(90)=102.22ms p(95)=102.36ms
+```
+
+When the disruption of an additional `100ms` delay and an error rate of 10% of requests is introduced for a period of `30s`, we can see how the stastics are affected (in particular, `p(90)` and `p(95)`):
```
Comment (dgzlopes) [inline README.md:192]:
```suggestion
When the disruption of an additional `100ms` delay and an error rate of 10% of requests is introduced for a period of `30s`, we can see how the statistics are affected (in particular, `p(90)` and `p(95)`):
```

File: agent/cmd/http.go [hunk 4]
```diff
@@ -0,0 +1,224 @@
...
+	"net/url"
+	"os/exec"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/spf13/cobra"
+)
+
+// delayCmd handles network delay command execution
```
Comment (dgzlopes) [inline agent/cmd/http.go:18]:
```suggestion
// httpCmd handles http delay command execution
```

File: examples/disrupt-http.js [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+
+  // delay traffic from one random replica of the deployment
+  const podDisruptor = new PodDisruptor(
+    k8sClient,
+    {
+      namespace: data.namespace,
+      selector: { app: app},
+      picker: 'random'
+    }
+  )
```
Comment (dgzlopes) [inline examples/disrupt-http.js:46]: new PodDisruptor().disruptHttp feels wrong 👀 

At least my mind crashes a bit with having disrupt in the method too, but yeah, big nit.

**Review by dgzlopes — APPROVED** [review]
LGTM!

I mostly omitted the Go part, as this is a POC, and it's not that critical. Added a small nit about the API and fixed some typos. Thanks for the work!

File: examples/disrupt-http.js (hunk 5 above)
Comment (pablochacin) [reply examples/disrupt-http.js:46]: What will you suggest? The method does potentially several things: delay requests, inject errors.

File: README.md (hunk 1 above)
Comment (pablochacin) [reply README.md:126]: Good catch. Thanks,

File: examples/disrupt-http.js (hunk 5 above)
Comment (dgzlopes) [reply examples/disrupt-http.js:46]: We instantiate a PodDisruptor, with some selectors, which are our targets. 

Then, maybe, it makes sense to pick the layer where the disruption happens. 

Example: `new PodDisruptor(<how we select>).http(<what we want to do to the layer>)`

Comment (dgzlopes) [reply examples/disrupt-http.js:46]: But yeah, proposal from the top of my mind, maybe it doesn't make sense 😄

Comment (pablochacin) [reply examples/disrupt-http.js:46]: Interesting. I was just thinking about some similar idea, but in the constructuror, like:
```
new PodDisruptor(
 client,
 kind,
 selector
 )
 ```
 where kind is `http`, `network`, `resources`, etc.

Comment (dgzlopes) [reply examples/disrupt-http.js:46]: I like the idea of picking the kind, not sure where it's the best place, but yeah!

Agree. Let's not focus too much on it right now 👍

**Pull request merged**